logdevice/common/test/NodesConfigurationPollerTest.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+#include "logdevice/common/NodesConfigurationPoller.h"
+
+#include <algorithm>
+
+#include <folly/Random.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "logdevice/common/ObjectPoller.h"
+#include "logdevice/common/RandomNodeSelector.h"
+#include "logdevice/common/configuration/Configuration.h"
+#include "logdevice/common/configuration/nodes/NodesConfigurationCodec.h"
+#include "logdevice/common/test/TestUtil.h"
+#include "logdevice/common/toString.h"
+
+using namespace facebook::logdevice;
+
+namespace {
+
+class MockNodesConfigurationPoller : public NodesConfigurationPoller {
+ public:
+  virtual ~MockNodesConfigurationPoller() {}
+
+  MockNodesConfigurationPoller(
+      Poller::Options options,
+      VersionExtFn version_fn,
+      Callback cb,
+      folly::Optional<u_int32_t> node_order_seed,
+      folly::Optional<Version> conditional_base_version)
+      : NodesConfigurationPoller(std::move(options),
+                                 std::move(version_fn),
+                                 std::move(cb),
+                                 node_order_seed,
+                                 std::move(conditional_base_version)) {}
+
+ protected:
+  Poller::RequestResult sendRequestToNode(Poller::RoundID round,
+                                          node_index_t node) override {
+    return Poller::RequestResult::OK;
+  }
+
+  std::shared_ptr<const configuration::nodes::NodesConfiguration>
+  getNodesConfiguration() const override {
+    auto node_config = createSimpleNodesConfig(1 /* 1 node */);
+    auto server_config = ServerConfig::fromDataTest(
+        "random_node_selector_test", std::move(node_config));
+
+    auto nodes = server_config->getNodes();
+    auto node_id =
+        NodeID(nodes.begin()->first, nodes.begin()->second.generation);
+    return server_config->getNodesConfigurationFromServerConfigSource();
+  }
+
+  MOCK_METHOD0(getMyNodeID, folly::Optional<node_index_t>());
+  MOCK_METHOD0(getClusterState, ClusterState*());
+  
+
+  // return true if the NCPoller is running in the bootstrapping environment
+  // (e.g., NodesConfigurationInit). If so, conditional polling will be disabled
+  bool isBootstrapping() const override {
+     return false;
+  }
+
+ public:
+  auto testFunction() {
+    return *buildPollingSet({1, 2, 3, 4, 5}, {}, {}, {}, 1, 0).begin();
```

Comment:
Maybe return the whole set here and compare equality of the whole set instead of only the first element? Also use maybe 3 required nodes instead of 1?

Reply:
done